atc/creds/ssm/manager_factory.go

```diff
@@ -26,6 +28,28 @@ func (factory *ssmManagerFactory) AddConfig(group *flags.Group) creds.Manager {
 	return manager
 }
 
-func (factory *ssmManagerFactory) NewInstance(interface{}) (creds.Manager, error) {
-	return &SsmManager{}, nil
+func (factory *ssmManagerFactory) NewInstance(config interface{}) (creds.Manager, error) {
+	if c, ok := config.(map[string]interface{}); !ok {
+		return nil, fmt.Errorf("invalid aws ssm config format")
+	} else {
+		manager := &SsmManager{
+			TeamSecretTemplate: DefaultTeamSecretTemplate,
+			PipelineSecretTemplate : DefaultPipelineSecretTemplate,
+		}
+
+		decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+			ErrorUnused: true,
+			Result:      &manager,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		err = decoder.Decode(c)
```

Comment:
I think this can just use `config` directly and we don't need the conversion check on lines 32-34 anymore. 👍